### DISCUSSION TITLE
Consulta diagrama de clases QMP

### DISCUSSION THREAD

**Discussion by juanmpereira** [dds-jv/foro#117] [Consultas de objetos] [answered]
Buenas, al volver a hacer el diagrama de clases de QMP me surgio la duda de como iria el borrador. Lo pongo como una entidad que apunta a la prenda?
 A su vez, a la clase preconditions la pongo como una entidad donde la prenda apunta a ella?

**Comment by juanmpereira** [comment]
Un agregado, todas las implementaciones de sastre van a llamar a borrador para crear la prenda; Tengo que hacer que cada una apunte a borrador o que sastre (la interfaz) apunte a borrador o de otra forma?

**Comment by gastonprieto** [comment, marked as answer]
> Buenas, al volver a hacer el diagrama de clases de QMP me surgio la duda de como iria el borrador. Lo pongo como una entidad que apunta a la prenda?

En cual de las dos opciones? en una tiene un atributo y debería ir la referencia. 
En la otra solución solo la instancia, así que podes hacer una flecha punteada que indique `<<crea>>`

```
------------         <<crea>>                --------------
| Borrador |   - - - - - - - - - - - - - -> |    Prenda   |
------------                                 --------------
```

> A su vez, a la clase preconditions la pongo como una entidad donde la prenda apunta a ella?

No hace falta agregarla al diagrama, porque no es tan relevante para explicar tu solución. En general pones las abstracciones mas relevantes para resolver todo el problema (si no es mucho) o un requerimiento en particular, incluso podes segmentar el diagrama de clases y tener varios según en lo que quieras hacer énfasis.

> todas las implementaciones de sastre van a llamar a borrador para crear la prenda; Tengo que hacer que cada una apunte a borrador o que sastre (la interfaz) apunte a borrador o de otra forma

Yo creo que lo haría en la interfaz y haría un comentario que explique porque lo pusiste ahí.

**Comment by juanmpereira** [comment]
Entendido, gracias!
